### PR TITLE
Build dynamic DLL version on Windows

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Currently, the library is build in `release-static` mode which means that it can't be linked into other DLL's. This PR changes the build to `release-dynamic` and copies the resulting DLL's (and import libs) over to the install location.

See https://github.com/conda-forge/gdal-feedstock/pull/346

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
-copy output\release-static\%ARCH%\lib\libwebp.lib %LIBRARY_PREFIX%\lib\libwebpdecoder.lib
+copy output\release-dynamic\%ARCH%\bin\cwebp.exe %LIBRARY_PREFIX%\bin\cwebp.exe
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\dwebp.exe %LIBRARY_PREFIX%\bin\dwebp.exe
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebp.dll %LIBRARY_PREFIX%\bin\libwebp.dll
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebpdecoder.dll %LIBRARY_PREFIX%\bin\libwebpdecoder.dll
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebpdemux.dll %LIBRARY_PREFIX%\bin\libwebpdemux.dll
```
Comment (gillins) [inline recipe/bld.bat]: Is there a reason why `libpdemux` was missed in the original recipe?

File: recipe/bld.bat [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
 :: Build!
-nmake /f Makefile.vc CFG=release-static RTLIBCFG=static OBJDIR=output
+nmake /f Makefile.vc CFG=release-dynamic RTLIBCFG=dynamic OBJDIR=output
 if errorlevel 1 exit 1
 
 :: Copy the dll's of these dependencies
-copy output\release-static\%ARCH%\bin\cwebp.exe %LIBRARY_PREFIX%\bin\cwebp.exe
-copy output\release-static\%ARCH%\bin\dwebp.exe %LIBRARY_PREFIX%\bin\dwebp.exe
-copy output\release-static\%ARCH%\lib\libwebp.lib %LIBRARY_PREFIX%\lib\libwebp.lib
-copy output\release-static\%ARCH%\lib\libwebp.lib %LIBRARY_PREFIX%\lib\libwebpdecoder.lib
```
Comment (gillins) [inline recipe/bld.bat]: I think this might have been a typo? `libwebpdecoder.lib` appears to be a distinct lib

File: recipe/bld.bat [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\dwebp.exe %LIBRARY_PREFIX%\bin\dwebp.exe
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebp.dll %LIBRARY_PREFIX%\bin\libwebp.dll
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebpdecoder.dll %LIBRARY_PREFIX%\bin\libwebpdecoder.dll
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\bin\libwebpdemux.dll %LIBRARY_PREFIX%\bin\libwebpdemux.dll
+if errorlevel 1 exit 1
+copy output\release-dynamic\%ARCH%\lib\libwebp_dll.lib %LIBRARY_PREFIX%\lib\libwebp.lib
```
Comment (gillins) [inline recipe/bld.bat]: Do we need to keep the same naming, or is `_dll` ok?

**Comment by gillins** [comment]
@isuruf any idea why `winres.h` isn't available with VS2008 on Azure? Should be installed with the SDK...

**Comment by isuruf** [comment]
I've no idea. Can you try appveyor?

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
Turns out it is called `winresrc.h` with VS2008....

**Comment by ocefpaf** [comment]
Merging this without aarch for now. Thanks @gillins!

**Pull request merged**